tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py

```diff
@@ -0,0 +1,206 @@
+import pytest
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow_addons.callbacks import AverageModelCheckpoint
+from tensorflow_addons.optimizers import MovingAverage
+import os
+
+try:
+    import h5py
+except ImportError:
+    h5py = None
+
+TRAIN_SAMPLES = 10
+NUM_CLASSES = 2
+INPUT_DIM = 3
+NUM_HIDDEN = 5
+BATCH_SIZE = 5
+EPOCHS = 5
+test_model_filepath = "test_model.h5"
```

Comment:
Using the same path for all the tests won't work if the tests are all run concurrently (with pytest-xdist for example). We also don't want to pollute the current working directory if one test fails and the file is not removed. The clean solution is to use this fixture: https://docs.pytest.org/en/latest/tmpdir.html#the-tmp-path-fixture , we also won't have to do `os.remove`.

Reply:
done